### PR TITLE
Make abstract player, random player, structure for evaluation of many games

### PR DESCRIPTION
**Pull request by aferron**
Creates abstract base class ChessPlayer requiring implementation of get_next_move(), and including a class exception

Makes RandomChessPlayer to randomly select the next move from the available legal moves

Moves running of game to new Game class, with public properties for board, terminated, and winner

Game class is instantiated with parameters of two players

Catches the case of no legal moves remaining and signals a draw

Alternates between players in the game class, ending when the game is won or is a draw

The class AIChess now manages multiple games and reports metrics on the game

### CODE REVIEW

File: aichess.py [hunk 1]
```diff
@@ -1,35 +1,43 @@
...
+            if winner is None:
+                draws += 1
 
-game = Game()
-game.run_example_game()
-game.board.print_binary_examples()
\ No newline at end of file
+            # If winner is True, white won (player1)
+            elif winner:
+                player1_wins += 1
```
Comment (nicolekurtz93) [inline aichess.py:33]: @aferron Should we track player2_wins, as well?

Comment (nicolekurtz93) [reply aichess.py:33]: ah never mind, I see below.

**Review by nicolekurtz93 — APPROVED** [review]

**Comment by WonkersTheWatilla** [comment]
Looks good to me!

**Review by rfilgas — APPROVED** [review]
Looks good.

**Pull request merged**